kahuna/public/js/search/index.js

```diff
@@ -129,7 +129,8 @@ search.config(['$stateProvider', '$urlMatcherFactoryProvider',
     });
 
     $stateProvider.state('search.results', {
-        url: 'search?{query:Query}&ids&since&nonFree&uploadedBy&until&orderBy',
+        url: 'search?{query:Query}&ids&since&nonFree&uploadedBy&until&orderBy' +
+             '&dateField&takenSince&takenUntil&modifiedSince&modifiedUntil',
```

Comment:
Nope, you can write your own types, but some are native.
I can't find an exhaustive list though.